### DISCUSSION TITLE
STM32F40X dual phy

### DISCUSSION THREAD

**Discussion by webbbn** [hathach/tinyusb#894] [Q&A] [answered]
Would tinyusb support two device phy at the same time? I was able to create a simple firmware with stm32cube that seems to work with both the FS device and HS device (configured as FS mode) in device mode at the same time, but I would much rather use tinyusb. I would like to create a device that has a WebUSB interface on one end, connected to one computer, and a keyboard/mouse HID device on the other end.

Is is possible to configure both phy on the STM32 using tinyusb on devices that support dual USB?

The other option might be a hybrid approach, where I run tinyusb on one phy to support WebUSB, and implement the HID interfaces with the lower-level stm32 libraries, but that seems like it could be a mess.

**Comment by hathach** [comment, marked as answer]
originally, it is designed to support multiple PHYs though the API is too complicated and requires additional rhport to all APIs and callbacks. And there is also no use for that. Therefore I have dropped that support from the API, it is still possible if you want to do the modification yourself. Short anwser is no, long answer is possible if you want to work it yourself.
